### DISCUSSION TITLE
No query results for model [App\Models\{model_name}] {method_name}

### DISCUSSION THREAD

**Discussion by ronaldaraujo** [laravel/framework#50214] [Q&A] [answered]
In the past, I saw a [tweet from Taylor](https://twitter.com/taylorotwell/status/1651593413140287488) where he talked about the 'best practices' of using the 7 'RESTful' methods. Since then, I have been applying this idea. However, now I have encountered this behavior. Look below:

`/routes/api.php`

Scenario 1: The application returned error `No query results for model [App\Models\{model_name}] {method_name}`

```php
...
Route::apiResource('receipts', ReceiptController::class)->names('api.receipts');

Route::delete('receipts/destroy-multiple', [DeleteMultipleController::class, 'destroy'])
            ->name('api.delete-multiple.destroy');
...
```

Scenario 2: The application works.

```php
...
Route::delete('receipts/destroy-multiple', [DeleteMultipleController::class, 'destroy'])
            ->name('api.delete-multiple.destroy');

Route::apiResource('receipts', ReceiptController::class)->names('api.receipts');
...
```

**My question: Is this behavior correct?**

**Comment by ryangjchandler** [comment, marked as answer]
@ronaldaraujo Yes, this is expected behaviour. The `apiResource` method registers something like `/receipts/{receipts}` which would match any `/receipts/*` requests before your more explicit `/receipts/destroy-multiple` route, therefore you need to register it before the API resource routes.
